tools/ReleaseNotes/data/whats-new-13.md

```diff
@@ -0,0 +1,1790 @@
+- **Efficient**: Only changed layers rebuild
+- **Consistent**: Same layers produce same results
+
+#### Real-World Example: Static Site with API
+
+A common pattern is building a frontend and serving it from a backend:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Build a Vite frontend in a container
+var frontend = builder.AddViteApp("frontend", "./frontend");
+
+// Python FastAPI backend
+var api = builder.AddUvicornApp("api", "./api", "main:app")
+    .WithUvEnvironment()
+    .WithExternalHttpEndpoints();
+
+// Extract frontend's /app/dist and copy to api's ./static
+api.PublishWithContainerFiles(frontend, "./static");
+
+builder.Build().Run();
+```
+
+When you deploy this:
+
+1. **Frontend container builds**: Vite builds the React/Vue/Svelte app inside a Node container
+2. **Files are extracted**: Aspire extracts `/app/dist` from the frontend container
+3. **Files are injected**: The dist files are copied into the API container at `./static`
+4. **Single deployment artifact**: The API container now contains both the Python app AND the frontend static files
+
+The FastAPI app can serve the static files:
+
+```python
+from fastapi import FastAPI
+from fastapi.staticfiles import StaticFiles
+
+app = FastAPI()
+
+# Serve the frontend static files
+app.mount("/", StaticFiles(directory="static", html=True), name="static")
+
+# API endpoints
+@app.get("/api/data")
+def get_data():
+    return {"message": "Hello from API"}
+```
+
+#### Advanced Usage: Multi-Stage Builds
+
+You can also use container files with .NET projects that produce build artifacts:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// .NET Blazor WebAssembly app that builds to wwwroot
+var blazorWasm = builder.AddProject<Projects.BlazorWasm>("blazor-wasm");
+
+// .NET API that serves static files
+var api = builder.AddProject<Projects.Api>("api");
+
+// Copy Blazor's published wwwroot into the API container
+api.PublishWithContainerFiles(blazorWasm, "./static");
+```
+
+#### Container Files in the Pipeline
+
+Container files integrate seamlessly with the Distributed Application Pipeline:
+
+- **Dependency tracking**: The pipeline knows that the API container depends on the frontend container
+- **Parallel execution**: Independent containers build in parallel
+- **Incremental builds**: Only changed containers rebuild
+- **Clear progress**: The dashboard shows which containers are building and extracting files
+
+This makes container files a natural fit for complex build workflows with multiple dependent services.
+
+#### Building Blocks: Container Files API
+
+The container files functionality is built on several key APIs that provide the building blocks for this paradigm:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var api = builder.AddProject<Projects.Api>("api")
+    .WithContainerFiles("./config", "/app/config", continueOnError: false)
```

Comment:
`WithContainerFiles` is actually something different and shouldn't be shown here.